manim/mobject/polyhedra.py

```diff
@@ -0,0 +1,360 @@
+    They are stored in the `graph` and `faces` attributes respectively.
+
+    .. manim:: PolyhedronSubMobjects
+        :save_last_frame:
+
+        class PolyhedronSubMobjects(ThreeDScene):
+            def construct(self):
+                self.set_camera_orientation(phi=75 * DEGREES, theta=30 * DEGREES)
+                octahedron = Octahedron(edge_length = 3)
+                octahedron.graph[0].set_color(RED)
+                octahedron.faces[2].set_color(YELLOW)
+                self.add(octahedron)
+    """
+
+    def __init__(
+        self,
+        vertex_coords: List[np.ndarray],
+        faces_list: List[List[Hashable]],
+        faces_config: dict = {},
+        graph_config: dict = {},
+    ):
+        VGroup.__init__(self)
+        self.faces_config = dict(
+            {"fill_opacity": 0.5, "shade_in_3d": True}, **faces_config
+        )
+        self.graph_config = dict(
+            {
+                "vertex_type": Dot3D,
+                "edge_config": {
+                    "stroke_opacity": 0,  # I find that having
+                },
+            },
+            **graph_config
+        )
+        self.vertex_coords = vertex_coords
+        self.vertex_indices = list(range(len(self.vertex_coords)))
+        self.layout = dict(enumerate(self.vertex_coords))
+        self.faces_list = faces_list
+        self.face_coords = [[self.layout[j] for j in i] for i in faces_list]
+        self.edges = self.get_edges(self.faces_list)
+        self.faces = self.create_faces(self.face_coords)
+        self.graph = Graph(
+            self.vertex_indices, self.edges, layout=self.layout, **self.graph_config
+        )
+        self.add(self.faces, self.graph)
+        self.add_updater(self.update_faces)
+
+    def get_edges(self, faces_list):
+        """Creates list of cyclic pairwise tuples."""
+        edges = []
+        for face in faces_list:
+            edges += zip(face, face[1:] + face[:1])
+        return edges
+
+    def create_faces(self, face_coords):
+        """Creates VGroup of faces from a list of face coordinates."""
+        face_group = VGroup()
+        for face in face_coords:
+            face_group.add(Polygon(*face, **self.faces_config))
+        return face_group
+
+    def update_faces(self, m):
+        face_coords = self.extract_face_coords()
+        new_faces = self.create_faces(face_coords)
+        self.faces.match_points(new_faces)
+
+    def extract_face_coords(self):
+        """Extracts the coordinates of the vertices in the graph.
+        Used for updating faces.
+        """
+        new_vertex_coords = []
+        for v in self.graph.vertices:
+            new_vertex_coords.append(self.graph[v].get_center())
+        layout = dict(enumerate(new_vertex_coords))
+        return [[layout[j] for j in i] for i in self.faces_list]
+
+
+class Tetrahedron(Polyhedron):
+    """A tetrahedron, one of the five platonic solids.
```

Comment:
It might be useful to note how many sides the Polyhedron has.
```suggestion
    """A tetrahedron, one of the five platonic solids, and one with four faces.
```
or
```suggestion
    """A tetrahedron, the polyhedron with four faces.
```

Reply:
I've added the vertex, edge and face information to the rest of the hardcoded polyhedra.